int32 x0
int32 y0
int32 x1
int32 y1
float32 prob
